packages/portal/src/Portal.tsx

```diff
@@ -1,80 +1,67 @@
-import { Component } from 'react';
+import { useEffect, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import { createPortal } from 'react-dom';
 
-interface PortalContainer {
-  el: Element;
-  remove: () => Element;
-}
-
-interface PortalProps {
+type PortalProps = {
   children?: React.ReactNode;
-  container?: HTMLElement;
-}
+} & (
+  | { container: HTMLElement; className?: never }
+  | { container?: never; className?: string }
+);
 
-interface PortalState {
-  defaultContainer: PortalContainer | null;
-}
+function createPortalContainer(className?: string): HTMLElement {
+  const el = document.createElement('div');
 
-export default class Portal extends Component<PortalProps, PortalState> {
-  static createPortalContainer(
-    nodeType: keyof HTMLElementTagNameMap = 'div',
-  ): PortalContainer {
-    const el = document.createElement(nodeType);
-    document.body.appendChild(el);
-
-    return {
-      el,
-      remove: () => document.body.removeChild(el),
-    };
+  if (className) {
+    el.className = className;
   }
+  document.body.appendChild(el);
 
-  static displayName = 'Portal';
+  return el;
+}
 
-  static propTypes = {
-    children: PropTypes.node,
-    container: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
-  };
+function Portal(props: PortalProps) {
+  const [container] = useState(
+    props.container ?? createPortalContainer(props.className),
+  );
 
-  state: PortalState = { defaultContainer: null };
+  const prevPropsRef = useRef(props);
 
-  componentDidMount() {
-    if (!this.props.container) {
-      this.setState({ defaultContainer: Portal.createPortalContainer() });
-    }
-  }
+  useEffect(() => {
+    const prevProps = prevPropsRef.current;
+    prevPropsRef.current = props;
 
-  shouldComponentUpdate(nextProps: PortalProps) {
-    if (this.props.container !== nextProps.container) {
+    if (
+      prevProps.container !== props.container ||
+      prevProps.className !== props.className
+    ) {
       // Sending consumer console error to control how this component is used
-      // and prevent unintended side-effects
       // eslint-disable-next-line no-console
       console.error(
-        'Changing the Portal container is not supported behavior and \
-        may cause unintended side effects. Instead, create a new \
-        Portal instance',
+        'Changing the Portal container or className is not supported behavior and \
+may cause unintended side effects. Instead, create a new Portal instance.',
       );
-      return false;
     }
+  });
 
-    return true;
-  }
+  useEffect(
+    () => () => {
+      if (!props.container) {
```

Comment:
If the component mounts without a container prop set, then this will always evaluate to true. Since we're passing in `[]` as the second argument to `useEffect` this will only be registered for the initial render function call.

Reply:
Right...but props is an object...so would the reference to `props.container` change if the value changes?

Honestly just creating a test for this case is all we need to ensure.